From: olh@suse.de
Subject: debug spurious 'Unknown symbol' errors in dmesg

the modules will work anyway.

	Bug 66533 - unknown symbols in usb modules
	*** Bug 63643 has been marked as a duplicate of this bug. ***
	*** Bug 64590 has been marked as a duplicate of this bug. ***
	*** Bug 66449 has been marked as a duplicate of this bug. ***

diff -purNx tags ../linux-2.6.11-rc5.modprobe_errors/kernel/module.c ./kernel/module.c
--- ../linux-2.6.11-rc5.modprobe_errors/kernel/module.c	2005-02-24 17:40:33.000000000 +0100
+++ ./kernel/module.c	2005-02-27 11:58:20.019987213 +0100
@@ -1157,8 +1157,8 @@ static int simplify_symbols(Elf_Shdr *se
 			if (ELF_ST_BIND(sym[i].st_info) == STB_WEAK)
 				break;
 
-			printk(KERN_WARNING "%s: Unknown symbol %s\n",
-			       mod->name, strtab + sym[i].st_name);
+			printk(KERN_WARNING "%s: Unknown symbol %s, st_info == 0x%x\n",
+			       mod->name, strtab + sym[i].st_name,(ELF_ST_BIND(sym[i].st_info)));
 			ret = -ENOENT;
 			break;
 
@@ -1723,6 +1723,8 @@ static struct module *load_module(void _
 	kfree(args);
  free_hdr:
 	vfree(hdr);
+	if (err)
+		printk(KERN_DEBUG "%s: err 0x%lx (dont worry)\n",__FUNCTION__, err);
 	if (err < 0) return ERR_PTR(err);
 	else return ptr;
 
